test-suite/src/foreign_key_on_update.rs

```diff
@@ -0,0 +1,221 @@
+//! Submodule for testing foreign key constraints on update.
+//!
+//! Since the Primary Keys cannot be updated at this time, operations such as `ON UPDATE CASCADE`,
+//! `ON UPDATE SET NULL` or `ON UPDATE SET DEFAULT` are not supported. The only supported operation
+//! is `ON UPDATE NO ACTION` or `ON UPDATE RESTRICT`. All other operations will result in a translation
+//! error.
+use {
+    crate::*,
+    gluesql_core::{
+        error::{AlterError, InsertError, TranslateError, UpdateError},
+        prelude::Payload,
+    },
+};
+
+test_case!(foreign_key_on_update, {
+    let g = get_tester!();
+
+    g.run(
+        "CREATE TABLE Referenced (
+            id INTEGER PRIMARY KEY,
+        );",
+    )
+    .await;
+
+    g.run(
+        "CREATE TABLE ReferencedNoPK (
+            id INTEGER
+        );",
+    )
+    .await;
+
+    g.named_test(
+        "On update operations on a table without primary key should fail",
+        "CREATE TABLE Referencing (
+            id INTEGER PRIMARY KEY,
+            referenced_id INTEGER,
+            FOREIGN KEY (referenced_id) REFERENCES ReferencedNoPK(id) ON UPDATE NO ACTION
+        );",
+        Err(AlterError::ReferencingNonPKColumn {
+            referenced_table: "ReferencedNoPK".to_owned(),
+            referenced_column: "id".to_owned(),
+        }
+        .into()),
+    )
+    .await;
+
+    g.named_test(
+        "On update cascade should fail as primary key cannot be updated",
+        "CREATE TABLE Referencing (
+            id INTEGER PRIMARY KEY,
+            referenced_id INTEGER,
+            FOREIGN KEY (referenced_id) REFERENCES Referenced(id) ON UPDATE CASCADE
+        );",
+        Err(TranslateError::UnsupportedConstraint("ON UPDATE CASCADE".to_owned()).into()),
```

Comment:
> What do you mean by Implement FK on UNIQUE ?

FK can be created on Column with Unique constraint as well.
And Unique column can be updated though PK can't.

So i want `cascade on unique` to be implemented.
1. Implement FK on UNIQUE
2. Write test cases of ON UPDATE CASCADE of ReferencedTableWithUnique
3. updating ReferencedTableWithUnique should update ReferencingTable as well.

Reply:
Isn't this something better suited for another PR?

Reply:
> Isn't this something better suited for another PR?

Yes, you'd better separate PR to implement FK on Unique !